wrangler-core/src/main/java/co/cask/directives/row/RecordRegexFilter.java

```diff
@@ -105,13 +101,11 @@ public void destroy() {
           }
         } else if (object instanceof String) {
           String value = (String) row.getValue(idx);
-          boolean matches = pattern.matcher(value).matches(); // pattern.matcher(value).matches();
-          if(!matched) {
-            matches = !matches;
-          }
-          if (matches) {
-            continue;
-          }
+          if (isMatching(value)) continue;
```

Comment:
Optimize code were you put isMatching in only one place. If so, then you don't need isMatching function.